src/libs/DebugUtils.ts

```diff
@@ -551,6 +590,75 @@ function validateReportActionJSON(json: string) {
     });
 }
 
+/**
+ * Gets the reason for showing LHN row
+ */
+function getReasonForShowingRowInLHN(report: OnyxEntry<Report>): TranslationPaths | null {
+    if (!report) {
+        return null;
+    }
+
+    const doesReportHaveViolations = OptionsListUtils.shouldShowViolations(report, transactionViolations);
+
+    const reason = ReportUtils.reasonForReportToBeInOptionList({
+        report,
+        // We can't pass report.reportID because it will cause reason to always be isFocused
+        currentReportId: '-1',
+        isInFocusMode: !!isInFocusMode,
+        betas,
+        policies,
+        excludeEmptyChats: true,
+        doesReportHaveViolations,
+        includeSelfDM: true,
+    });
+
+    // When there's no specific reason, we default to isFocused since the report is only showing because we're viewing it
+    if (reason === null || reason === CONST.REPORT_IN_LHN_REASONS.DEFAULT) {
+        return 'debug.reasonVisibleInLHN.isFocused';
+    }
+
+    return `debug.reasonVisibleInLHN.${reason}`;
+}
+
+/**
+ * Gets the reason that is causing the GBR to show up in LHN row
+ */
+function getReasonForShowingGreenDotInLHNRow(report: OnyxEntry<Report>, parentReportAction?: ReportAction): TranslationPaths | null {
```

Comment:
parentReportAction is unnecessary

Reply:
@pac-guerreiro Let's combine getGBRReportAction and getReasonForShowingGreenDotInLHNRow to one

Reply:
We only need to call getReasonAndReportActionThatRequiresAttention one time to get both reason and reportAction